src/scripts/tbtcv2-rewards/rewards.ts

```diff
@@ -172,12 +172,30 @@ export async function calculateRewards() {
   console.log(
     "Fetching AuthorizationDecreased events after rewards interval..."
   )
-  const postIntervalAuthorizationDecreasedEvents =
-    await tokenStaking.queryFilter(
-      "AuthorizationDecreaseApproved",
-      endRewardsBlock,
-      currentBlockNumber
-    )
+  let postIntervalAuthorizationDecreasedEvents = await tokenStaking.queryFilter(
+    "AuthorizationDecreaseApproved",
+    endRewardsBlock,
+    currentBlockNumber
+  )
+
+  // Special case: Dec 1st 23 distribution - legacy stakes
+  const legacyEvents = await getLegacyEvents(provider)
+
+  const intervalLegacyEvents = legacyEvents.filter(
+    (event) =>
+      event.blockNumber >= startRewardsBlock &&
+      event.blockNumber < endRewardsBlock
+  )
+  const postLegacyEvents = legacyEvents.filter(
+    (event) =>
+      event.blockNumber > startRewardsBlock &&
+      event.blockNumber > endRewardsBlock - 1
```

Comment:
This shouldn't be just:
```
event.blockNumber >= endRewardsBlock
```
?

Why do we need to check `startRewardsBlock` here?